build/Dockerfile

```diff
@@ -8,5 +8,6 @@ RUN make prep build strip
 FROM registry.access.redhat.com/ubi8/ubi-minimal
 COPY --from=build /build/pbapi /pbapi
 COPY --from=build /build/pbmigrate /pbmigrate
+COPY --from=build /build/api /api
```

Comment:
Can you use embed package instead? We do the same for migrations already to avoid having to copy additional files with the binary, it's really easy to use specifically for exposing a web directory:

https://pkg.go.dev/embed

Reply:
From the documentation:

_The //go:embed directive accepts multiple space-separated patterns for brevity, but it can also be repeated, to avoid very long lines when there are many patterns. The patterns are **interpreted relative to the package** directory containing the source file. The path separator is a forward slash, even on Windows systems. Patterns **may not contain ‘.’ or ‘..’** or empty path elements, nor may they begin or end with a slash. To match everything in the current directory, use ‘*’ instead of ‘.’. To allow for naming files with spaces in their names, patterns can be written as Go double-quoted or back-quoted string literals._

You need to have the Go source file in the same directory as the files you want to embed, that is the limitation. But that's doable, we can drop a file in the /api directory that's not big deal. Simply change this to:

```go
// api/embed.go

package api

import (
	"embed"
	"net/http"
)

//go:embed *
var embeddedSpecs embed.FS

func ServeOpenAPISpec(w http.ResponseWriter, r *http.Request) {
	http.FileServer(http.FS(embeddedSpecs))
}
```

Then use this in the main or all_routes.go file by simply importing the api via:

```go

import (
 "github.com/RHEnVision/provisioning-backend/api"
)

// api.ServeOpenAPISpec()
```